proposals/0082-relax-transaction-constraints.md

```diff
@@ -0,0 +1,173 @@
+
+The intent with relaxing these constraints is to minimize the amount of
+account-state which is required in order to validate a block. This gives more
+flexibility to when and how account-state is updated during both block
+production and validation.
+
+With these constraints removed, there is still one account-state constraint
+at the transaction level: address lookup table resolution.
+With this proposal, this constraint is intentionally not relaxed since it is
+necessary for validation of entry-level and block-level constraints.
+However, if/when those constraints are relaxed, this constraint should be
+relaxed as well.
+
+The relaxation of these constraints is only relaxed for transactions being
+included in a block. During block validation, if any of these constraints
+are broken, the entire block is marked invalid; with this proposal, the
+block is not marked invalid, but the transaction will not be executed.
+These constraints must still be satisfied in order for the transaction to be
+executed, and have an effect on state. However, there are some new considerations
+which must be taken into account, specifically as it relates to fees and block
+limits.
+
+### Fee-Paying
+
+Currently, iff a transaction's fee-payer does not have enough funds to pay the
+fee, the transaction cannot be included in a block. With this proposal, it is
+possible for such a transaction to be included in the block, and there are
+three different edge-cases to consider:
+
+1. The fee-payer account does not exist (0 lamports)
+2. The fee-payer account does not have enough funds for the entire fee
+3. The fee-payer account has enough funds for the fee, but would no longer be
+rent-exempt
+
+In case 1, the transaction should simply be ignored for execution, and have no
+effect on state.
+In case 2, the fee-paying account should be drained of all funds, but have no
+other effect on state.
+In case 3, the fee-paying account should be drained of all funds, with fees
+being paid to the block-producer, and the remainder of lamports being dropped.
+
+### Block-Limits
+
+Pending the activation of `2ry7ygxiYURULZCrypHhveanvP5tzZ4toRwVp89oCNSj`,
+validators must validate a block is within block-limits.
+With this proposal, some transactions may not be executable, but will still
+count towards block-limits.
+If these transactions did not count towards block-limits, the validation of
+block-limits would require the validator to check whether or not a transaction
+is executable, which negates the benefits of this proposal.
+
+Additionally, f these transactions did not count towards block-limits, a
+malicious leader could produce a block with non-executable transactions and
+overload the network.
+
+## Impact
+
+- Transactions that would previously be dropped with an error, can now be
+  included, and even charged fees.
+  - Users must be more careful when constructing transactions to ensure they
+    are executable if they don't want to waste fees
```

Comment:
Simulation doesn't change. It currently would tell you that your transaction had an error. That error doesn't change, it only changes what will happen to a block it may be included in.

For example, if a transaction had an invalid program account, simulation will still return `ProgramAccountNotFound` error.
If the user simulates, and then still submits that transaction, then they probably deserve to be charged, imo.